components/ActionSheet/src/private/MDCActionSheetHeaderView.m

```diff
@@ -0,0 +1,253 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCActionSheetHeaderView.h"
+
+#import "MaterialMath.h"
+#import "MaterialTypography.h"
+
+static const CGFloat TitleLabelAlpha = 0.87f;
+static const CGFloat MessageLabelAlpha = 0.6f;
+static const CGFloat MessageOnlyPadding = 23.f;
+static const CGFloat LeadingPadding = 16.f;
+static const CGFloat TopStandardPadding = 16.f;
+static const CGFloat TrailingPadding = 16.f;
+static const CGFloat TitleOnlyPadding = 18.f;
+static const CGFloat MiddlePadding = 8.f;
+
+@interface MDCActionSheetHeaderView ()
+
+@property(nonatomic, nonnull, strong) UIScrollView *scrollView;
+
+@end
+
+@implementation MDCActionSheetHeaderView {
+  UILabel *_titleLabel;
+  UILabel *_messageLabel;
+}
+
+@synthesize mdc_adjustsFontForContentSizeCategory = _mdc_adjustsFontForContentSizeCategory;
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    self.scrollView = [[UIScrollView alloc] initWithFrame:CGRectZero];
```

Comment:
Why does the header view need a scroll view?

Reply:
It doesn't need it as of now, but I think later it may for dynamic type. I don't know if an app bar is going to be needed or not. This can be removed in the current implementation but will possibly be added back later.

Reply:
More context: I'm concerned about the possibility of the header being scrollable separately from the content (which is its own scroll view). I'd expect the header to be sized to fits its contents and for the table view to be positioned below the header accordingly.

Reply:
The scroll view in current implementation is the size of it's content and no scrollable.

Reply:
I see. If possible, let's remove the scroll view and add it back if we find that we need it (I'm not certain that that will be the case).